### DISCUSSION TITLE
Failed to load .env file

### DISCUSSION THREAD

**Discussion by abhi3700** [DioxusLabs/dioxus#3163] [Q&A] [answered]
Hi!
I was trying to load the .env file using [dotenv](https://crates.io/crates/dotenv) crate. But somehow it was not getting set. I am not sure what i am missing.

- [main.rs](https://github.com/abhi3700/dioxus-playground/blob/7c79fc11c3382e5828ae985d524aecab51617b4d/src/main.rs).
- Want to use it inside [img_gen.rs](https://github.com/abhi3700/dioxus-playground/blob/7c79fc11c3382e5828ae985d524aecab51617b4d/src/img_gen.rs) which is one of the routes.

What am I missing?
Anyone please?

**Comment by wiseaidev** [comment, marked as answer]
You can access these env vars on the server (e.g., Axum) when using Dioxus full-stack, but not on the client side (Dioxus Web) in the browser. I had the same question while building [this full-stack blog app](https://github.com/opensass/opensass). I needed a way to set the admin email and password in the environment variables. Then I realized I could [check the user's role for login](https://github.com/opensass/opensass/blob/4d223a0c6829ab71be374de7e3e202c59f2bceec/src/pages/login.rs#L52-L55), setting it to 'admin' by manually updating the role in MongoDB from the default 'user' to 'admin.'

Edit: `dotenv` sets env vars at runtime and can't be accessed from WASM env. Try using [the `std::env` macro](https://doc.rust-lang.org/std/macro.env.html) to access these env vars at compile time, wasm.

**Reply by abhi3700** [reply]
I did that. Got resolved.

**Reply by wiseaidev** [reply]
> I did that. Got resolved.

Awesome!

**Comment by abhi3700** [comment]
Was taken up in the issue as well with some code snippets.
https://github.com/DioxusLabs/dioxus/issues/3164

**Comment by iesahin** [comment]
When you add environment secrets to the compiled WASM code in the compile time with `env!`, doesn't this send all your secrets to the browser? If someone decompiles WASM modules, they can get the secrets, right? Am I missing something here?

**Reply by ealmloff** [reply]
Just the value of the environment variable. It doesn't capture your whole environment. If you have secrets you don't want to leak, they need to stay on the server. For some API keys that are tied to a specific domain, exposing them on the client may be ok. Eg. a [google maps API key](https://developers.google.com/maps/documentation/javascript/get-api-key#restrict_key) could be fine to leak on the client if you scope it to a specific domain
